### PR TITLE
Enhancement/add pagination of teacher comments

### PR DESCRIPTION
**Pull request by SharpDevOps10**

### CODE REVIEW

File: src/v2/api/services/PollService.ts [hunk 1]
```diff
@@ -119,9 +139,7 @@ export class PollService {
               },
             },
           },
-          orderBy: query.sortBy
-            ? CommentsSortMapper[query.sortBy]
-            : undefined,
+          orderBy: query.sortBy ? CommentsSortMapper[query.sortBy] : undefined,
```
Comment (basarabst) [inline src/v2/api/services/PollService.ts:142]: previous version is more readable

**Review by basarabst — CHANGES_REQUESTED** [review]
Because of your changes, the function returns a completely empty array with no information about the question, this will need fixing

**Review by basarabst — APPROVED** [review]

**Review by Hahhhayam — DISMISSED** [review]
image
don`t throw an exeption when user set unexisting subjectId

**Review by Hahhhayam — APPROVED** [review]

File: src/v2/api/dtos/CommentsQueryDTO.ts [hunk 2]
```diff
@@ -57,4 +57,9 @@ export class CommentsQueryDTO {
   @IsEnum(CommentsSort)
   @IsOptional()
     sortBy?: CommentsSort;
+  @IsOptional()
```
Comment (just-hoshinon) [inline src/v2/api/dtos/CommentsQueryDTO.ts:60]: Should be shown in the swagger

File: src/v2/api/services/PollService.ts [hunk 3]
```diff
@@ -137,9 +155,14 @@ export class PollService {
...
           },
+          ...pageOptions,
         },
       },
-    }) as unknown as Promise<DbQuestionWithDiscipline[]>;
+    };
+
+
+
+    return await this.questionRepository.findMany(questionFindManyArgs) as DbQuestionWithDiscipline[];
```
Comment (just-hoshinon) [inline src/v2/api/services/PollService.ts:165]: excessive await

**Review by just-hoshinon — APPROVED** [review]

**Pull request merged**